src/GrumPHP/Task/Phpparser.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace GrumPHP\Task;
+
+use GrumPHP\Task\AbstractParserTask;
+use GrumPHP\Exception\RuntimeException;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+
+/**
+ * Php Parser task
+ */
+class Phpparser extends AbstractParserTask
+{
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return 'php_parser';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = parent::getConfigurableOptions();
+        $resolver->setDefault('triggered_by', array('php'));
+
+        return $resolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canRunInContext(ContextInterface $context)
+    {
+        return ($context instanceof GitPreCommitContext || $context instanceof RunContext);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function run(ContextInterface $context)
```

Comment:
We changed the `run()` method It now ALWAYS has to return a `TaskResultInterface`:
https://github.com/phpro/grumphp/blob/master/src/GrumPHP/Task/TaskInterface.php

We aren't relying on exceptions anymore. There are 3 statuses available: passed, failed, skipped